### PR TITLE
signup

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: "from@example.com"
+  default from: ENV["SMTP_email"]
```
Comment (thanhhungchu95) [inline app/mailers/application_mailer.rb]: SMTP_EMAIL

File: app/models/user.rb [hunk 2]
```diff
@@ -22,7 +45,28 @@ def self.new_token
...
   end
+
+  def send_activation_email
+    UserMailer.account_activation(self).deliver_now
+  end
+
+  private
+
+  def downcase_email
+    self.email = email.downcase
```
Comment (thanhhungchu95) [inline app/models/user.rb]: dùng downcase!

**Comment by thanhhungchu95** [comment]
(commented)

File: app/controllers/account_activations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  def edit
+    user = User.find_by email: params[:email]
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
+      user.update_attributes activated: true, activated_at: Time.zone.now
+      login user
+      flash[:success] = t ".acount_success"
+      redirect_to root_url
+    else
+      flash[:danger] = t ".acount_danger"
+      redirect_to root_url
```
Comment (ghost) [inline app/controllers/account_activations_controller.rb]: lặp code

File: app/views/user_mailer/account_activation.text.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Hi <%= @user.full_name %>,
```
Comment (ghost) [inline app/views/user_mailer/account_activation.text.erb]: i18n

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -1,42 +1,42 @@
 <div class="container">
   <div id="content">
-    <form action="#" method="post" class="beta-form-checkout">
+    <%= form_for @user,url: signup_path,
```
Comment (ghost) [inline app/views/users/new.html.erb]: space

File: config/routes.rb [hunk 6]
```diff
@@ -13,8 +13,11 @@
   get "/contact", to: "static_pages#contact"
   get "/checkout", to: "checkout#index"
   get "/signup", to:"users#new"
+  post "/signup", to:"users#create"
   get "/shopping_cart", to: "cart#shopping_cart"
   get "search(/:search)", to: "search#index" , as: :search
   root "products#index"
   resources :products
+  resources :users, only: %i(new create)
+  resources :account_activations, only: [:edit]
```
Comment (ghost) [inline config/routes.rb]: `only: :edit`

**Pull request merged**